### PR TITLE
build: add devcontainer setup

### PR DESCRIPTION
**Pull request by havogt**
Add devcontainer configuration with special customizations for VS Code.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM mcr.microsoft.com/devcontainers/python:1-3.10-bookworm
+RUN apt-get update \
+    && export DEBIAN_FRONTEND=noninteractive && apt-get install -y libboost-dev \
+    && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
```
Comment (egparedes) [inline .devcontainer/Dockerfile:4]: Can we also add some basic python tooling like `pipx` & `uv`? For example:
```suggestion
    && export DEBIAN_FRONTEND=noninteractive && apt-get install -y libboost-dev \
    && apt-get install pipx && ensurepath && pipx install uv \
    && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
```

File: .devcontainer/setup.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+ln -sfn /workspaces/gt4py/.devcontainer/.vscode /workspaces/gt4py/.vscode
+python -m venv .venv
+source .venv/bin/activate
+pip install --upgrade pip setuptools wheel
+pip install -e .
+pip install -r requirements-dev.txt
+pip install -i https://test.pypi.org/simple/ atlas4py
```
Comment (egparedes) [inline .devcontainer/setup.sh:9]: If we install `uv` in the dockerfile then we can use it here which should be nicer and much faster:
```suggestion
uv venv .venv
source .venv/bin/activate
uv pip install -r requirements-dev.txt
uv pip install -e .
uv pip install -i https://test.pypi.org/simple/ atlas4py
```

**Review by egparedes — COMMENTED** [review]
Just a couple of suggestions to use `uv`:

**Comment by havogt** [comment]
Should we just `--user` or `--system` install instead of venv? Since the devcontainer is anyway an isolated environment...

**Comment by egparedes** [comment]
> Should we just `--user` or `--system` install instead of venv? Since the devcontainer is anyway an isolated environment...

Yes, I agree. I don't think we need a venv inside the container

**Comment by egparedes** [comment]
@havogt Is there any blocker to get this updated and merged? I'd like to start using it soon.

**Comment by egparedes** [comment]
I've added some vscode extensions and settings to the proper customization point in devcontainer.json. It looks good to me so I'm approving the PR but you should also take a look to my changes and modify then as you wish.

**Review by egparedes — APPROVED** [review]
LGTM

**Pull request merged**